### PR TITLE
Add deletion to gift cards

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it adds deletion to gift cards

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://feature-gift-cards-mvp.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3906 Add deletion and modal to Gift Cards list](https://app.clickup.com/t/2549495/SALEOR-3906)

File: src/giftCards/GiftCardsList/providers/GiftCardListProvider/hooks/useGiftCardListBulkActions.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import {
+  GiftCardsListConsumerProps,
+  GiftCardsListContext
+} from "../GiftCardListProvider";
+
+export type UseGiftCardListBulkActionsProps = UseBulkActionsProps &
+  Pick<GiftCardsListConsumerProps, "selectedItemsCount">;
+
+const useGiftCardListBulkActions = (): UseGiftCardListBulkActionsProps => {
```
Comment (jwm0) [inline src/giftCards/GiftCardsList/providers/GiftCardListProvider/hooks/useGiftCardListBulkActions.ts:12]: If these are separate hooks, doesn't it also make sense to split the providers as well? It's important we don't bloat the contexts too much

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/giftCards/GiftCardsList/GiftCardsListHeader.tsx [hunk 2]
```diff
@@ -7,16 +7,13 @@ import React from "react";
 import { useIntl } from "react-intl";
 
 import { giftCardsListHeaderMenuItemsMessages as messages } from "./messages";
+import useGiftCardListDialogs from "./providers/GiftCardListDialogsProvider/hooks/useGiftCardListDialogs";
```
[the remark this answers is not in the document]
Comment (mmarkusik) [reply src/giftCards/GiftCardsList/GiftCardsListHeader.tsx:10]: Will make a separate pr for refactors 👍

File: src/giftCards/GiftCardsList/providers/GiftCardListProvider/hooks/useGiftCardListBulkActions.ts (hunk 1 above)
Comment (mmarkusik) [reply src/giftCards/GiftCardsList/providers/GiftCardListProvider/hooks/useGiftCardListBulkActions.ts:12]: Will make a separate pr for refactors 👍

**Review by orzechdev — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**